proposals/NNNN-GetInputScreen.md

```diff
@@ -0,0 +1,140 @@
+      An image icon for the input field
+    </description>
+  </param>
+  <param name="text" maxlength="500" type="String" mandatory="false">
+    <description>
+      The text that is already present in the input field
+    </description>
+  </param>
+  <param name="format" maxlength="500" type="String" mandatory="false">
+    <description>
+      A text string used to format the input text to a certain style. Any characters except underscore will remain in the field with no way to type over them. The underscore character will represent a space that the user can type in.  E.g. A format string for Phone number would be +_(___)___-____
+    </description>
+  </param>
+  <param name="additionalInfo" maxlength="500" type="String" mandatory="false">
+    <description>
+      More info that is shown by the input field
+    </description>
+  </param>
+  <param name="error" type="Boolean" mandatory="false">
+    <description>
+      If set to true, then the field has some problem with it and the HMI should highlight it in red.
+    </description>
+  </param>
+  <param name="comboBoxChoices" type="String" maxlength="500" array="true" minsize="0" maxsize="200" mandatory="false">
+    <description>
+      List of choices if the type is COMBO_BOX. Otherwise ignored.
+    </description>
+  </param>
+  <param name="inputMask" type="KeyboardInputMask" mandatory="false">
+    <description>
+      Used to cover up personal information like credit card number. Defaults to "DISABLE_INPUT_KEY_MASK"
+    </description>
+  </param>
+</struct>
+```
+```xml
+  <enum name="FieldType" since="X.X">
+    <description>The type of input field</description>
+    <element name="COMBO_BOX" />
+    <element name="TEXT" />
+    <element name="NUMBER" />
+      <description>
+        This is a notice for the HMI to bring up a keyboard, however the response for this field will still be a string.
+      </description>
+  </enum>
+```
+
+And the response which would be sent when the user presses a softbutton - this would use RPC encryption which is handled by policies.
+One string response for each field. If a field was left empty, the associated string would be blank.
+```xml
+<function name="GetInput" messagetype="response">
+  <param name="fields" type="String" maxlength="500" array="true" minsize="0" maxsize="100" mandatory="true">
+    <description>
+      One string response for each field. If a field was left empty, the associated string would be blank. If the field had formatting characters, they are included in the response as well. 
+    </description>
+  </param>
+</function>
+```
+
+We can use the `maskInputCharacters` parameter from [0238-Keyboard-Enhancements](https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0238-Keyboard-Enhancements.md) for password or other sensitive data input.
+
+## Potential downsides
+
+Added complexity for Core and HMI
+
+## Impact on existing code
+
+It's a new parameter so this would not affect older SDL versions.
```

Comment:
Can you please advise how each library would be impacted?  For example, will this require a minor version change for each listed in `Impacted platforms`?

Reply:
@MichaelCrimando it looks like you added this to the `Potential downsides` section instead of `Impact on existing code`.  Was that intended?  Based on your update, I think `HMI` will also need to be added as an Impacted Platform at the top of the proposal.